springExercise/src/main/java/com/tekcamp/springExercise/Controller/UserController.java

```diff
@@ -0,0 +1,109 @@
+package com.tekcamp.springExercise.Controller;
+
+import com.tekcamp.springExercise.Model.Request.UserRequest;
+import com.tekcamp.springExercise.Model.Response.UserResponse;
+import com.tekcamp.springExercise.Model.User;
+import com.tekcamp.springExercise.Services.UserService;
+import com.tekcamp.springExercise.Shared.Dto.UserDto;
+import org.springframework.beans.BeanUtils;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("users")
+public class UserController {
+
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping
+    public ResponseEntity getUsers(@RequestParam(value = "page", defaultValue = "1" ) int page,
+                                   @RequestParam(value = "limit", defaultValue = "3" ) int limit){
+        //Getting a list of users from the db
+        List<UserDto> usersList = userService.getUsers(page, limit);
+        //Creating a list to store the response in
+        List<UserResponse> returnList = new ArrayList<>();
+        if(usersList.isEmpty()) {
+            return ResponseEntity.notFound().build();
+        }
+        //Creating a clean response list with no sensitive information and returning it
+        usersList.forEach(user -> {
+            UserResponse userCopy = new UserResponse();
+            BeanUtils.copyProperties(user, userCopy);
+            returnList.add(userCopy);
+        });
+        return ResponseEntity.accepted().body(returnList);
+    }
+
+    @GetMapping(path = "/id/{id}")
+    public ResponseEntity getUserById(@PathVariable Long id){
+        //Sending the id to search for a valid match
+        UserDto foundUser = userService.getUserById(id);
+        if(foundUser == null) {
+            return ResponseEntity.notFound().build();
+        }
+        //Creating a clean response with no sensitive information and returning it
+        UserResponse userCopy = new UserResponse();
+        BeanUtils.copyProperties(foundUser, userCopy);
+        return ResponseEntity.accepted().body(userCopy);
+    }
+
+    @GetMapping(path = "/{email}")
```

Comment:
Don't expose email in a URI, that's private. User a public id and get the email from the request body